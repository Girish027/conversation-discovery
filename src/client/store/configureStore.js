import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'connected-react-router/immutable';
import {
  fromJS,
} from 'immutable';
import createRootReducer from './reducers';

const loggerMiddleware = createLogger({
  predicate: () => process.env.NODE_ENV === 'development',
  stateTransformer: (state) => JSON.parse(JSON.stringify(state)),
  actionTransformer: (action) => JSON.parse(JSON.stringify(action)),
});


const configureStore = (initialState = {}, history) => {
  const middlewares = [
    routerMiddleware(history), // for dispatching history actions
    thunk,
    loggerMiddleware
  ];

  const store = createStore(
    createRootReducer(history),
    fromJS(initialState),
    composeWithDevTools(applyMiddleware(...middlewares))
  );

  // Hot reloading
  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createRootReducer(history));
    });
  }

  return store;
};

export default configureStore;
