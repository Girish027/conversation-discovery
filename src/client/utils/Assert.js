/**
 * 24/7 Customer, Inc. Confidential, Do Not Distribute. This is an
 * unpublished, proprietary work which is fully protected under
 * copyright law. This code may only be used pursuant to a valid
 * license from 24/7 Customer, Inc.
 */

/**
 * Assertions module
 * @module helpers/assert
 */

const RX_EMPTY = /^\s*$/;

/**
 * Callback invoked when an assertion fails.
 * The default implementation throws an exception that includes a message describing the failed assertion.
 * @func _onFailure
 * @private
 * @memberOf module:helpers/assert
 * @param {String} msg Message to be shown when exception is thrown
 */
function onFailure(msg) {
  throw new Error(msg);
}

class Assert {
  /**
   * Assert a condition and throw an error if the condition is equivalent to `false`.
   * @func assert
   * @public
   * @memberOf module:helpers/assert
   * @param {Boolean} cond A condition to evaluate.
   * @param {String} [msg] The error message to throw.
   */
  static assert(cond, msg = 'assert failed') {
    if (!cond) {
      onFailure(msg);
    }
  }

  /**
   * Assert the given value to be an object.
   * @func isObject
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} obj An object to evaluate.
   * @param {String} [name] The name of the object.
   */
  static isObject(obj, name) {
    const msg = (name ? `${name} is not an object` : 'object expected');
    Assert.assert(typeof obj === 'object' && obj !== null, msg);
  }

  /**
   * Assert the given value to be undefined or null.
   * @func isUndefinedOrNull
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} obj An object to evaluate.
   * @param {String} [name] The name of the object.
   */
  static isUndefinedOrNull(obj, name) {
    const msg = (name ? `${name} is not undefined or null` : 'undefined or null expected');
    Assert.assert(obj === undefined || obj === null, msg);
  }

  /**
   * Assert the given value to be not undefined or null.
   * @func isNotUndefinedOrNull
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} obj An object to evaluate.
   * @param {String} [name] The name of the object.
   */
  static isNotUndefinedOrNull(obj, name) {
    const msg = (name ? `${name} is undefined or null` : 'undefined or null not expected');
    Assert.assert(obj !== undefined && obj !== null, msg);
  }

  /**
   * Assert the given value to be a boolean.
   * @func isBoolean
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} bool A boolean to evaluate.
   * @param {String} [name] The name of the boolean.
   */
  static isBoolean(bool, name) {
    const msg = (name ? `${name} is not a boolean` : 'boolean expected');
    Assert.assert(typeof bool === 'boolean', msg);
  }

  /**
   * Assert the given value to be a function.
   * @func isFunction
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} func A function to evaluate.
   * @param {String} [name] The name of the function.
   */
  static isFunction(func, name) {
    const msg = (name ? `${name} is not a function` : 'function expected');
    Assert.assert(typeof func === 'function', msg);
  }

  /**
   * Assert the given value to be a number.
   * @func isNumber
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} num A number to validate.
   * @param {String} [name] The name of the variable that stores the number.
   */
  static isNumber(num, name) {
    const msg = (name ? `${name} is not a number` : 'number expected');
    Assert.assert(typeof num === 'number', msg);
  }

  /**
   * Assert the given value is a string and not empty.
   * @func notEmpty
   * @public
   * @memberOf module:helpers/assert
   * @param {String} str A string to evaluate.
   * @param {String} [name] The name of the string.
   */
  static notEmpty(str, name) {
    const msg = (name ? `${name} is not a string or is empty` : 'non-empty string expected');
    Assert.assert(typeof str === 'string' && !RX_EMPTY.test(str), msg);
  }

  /**
   * Assert the given object has the given constructor.
   * @func instanceOf
   * @public
   * @memberOf module:helpers/assert
   * @param {Object} obj An object to evaluate.
   * @param {Object.constructor} constructor A constructor to evaluate.
   * @param {String} [className] The name of the class/constructor.
   */
  static instanceOf(obj, constructor, className) {
    const msg = (className ? `expected instance of ${className}` : 'unexpected instance type');
    Assert.assert(obj instanceof constructor, msg);
  }

  /**
   * Assert the given value to be a non-empty string, a number, or a boolean.
   * @func isScalar
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} value A variant to evaluate.
   * @param {String} [name] The name of the variable that stores the variant.
   */
  static isScalar(value, name) {
    const vType = typeof value;
    const msg = `${name ? `${name} ` : ''}must be string, number, or boolean`;
    Assert.assert(vType === 'string' || vType === 'number' || vType === 'boolean', msg);
    if (vType === 'string') {
      Assert.notEmpty(value, name);
    }
  }

  /**
   * Assert the given value to be an array.
   * @func isArray
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} value A variant to evaluate.
   * @param {String} [name] The name of the variable that stores the variant.
   */
  static isArray(value, name) {
    const msg = (name ? `${name} is not an array` : 'array expected');
    Assert.assert(Object.prototype.toString.call(value) === '[object Array]', msg);
  }
}


export default Assert;
