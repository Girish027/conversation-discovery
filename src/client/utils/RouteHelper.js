import { basePath } from '../swagger.json';

const baseRoute = `${basePath}/ui`;

const routeParams = ({
  clientId, appId, accountId, projectId, runId, clusterId, componentClientId
}) => {
  let params = `?clientid=${clientId}&accountid=${accountId}&appid=${appId}&componentClientId=${componentClientId}`;
  if (projectId) {
    params += `&project=${projectId}`;
    if (runId) {
      params += `&run=${runId}`;
      if (clusterId) {
        params += `&cluster=${clusterId}`;
      }
    }
  }

  return params;
};

export const routeNames = {
  BASE_ROUTE: baseRoute,
  HOME_PAGE: basePath,
  DISCOVER_INTENTS: `${baseRoute}/discover-intents`,
  DISTRIBUTION_GRAPH: `${baseRoute}/run-overview`,
  TOPIC_REVIEW: `${baseRoute}/topic-review`
};

export const getRoute = (routeName, params = {}) => routeName + routeParams(params);
