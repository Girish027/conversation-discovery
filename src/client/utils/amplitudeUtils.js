// import amplitude from 'amplitude-js/amplitude';

export const initializeAnalytics = (analyticsKey, userName = '') => {
  // const analytics = amplitude.getInstance();
  // analytics.init(analyticsKey, userName);
};

export const logAmplitudeEvent = (eventName, eventSpecificData = {}) => {
  // amplitude.getInstance().logEvent(eventName, {
  //   ...eventSpecificData,
  // });
};
