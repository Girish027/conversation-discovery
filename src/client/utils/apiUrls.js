import { basePath } from '../swagger.json';

const root = (process.env.NODE_ENV === 'production'
  ? '//'.concat(window.location.host)
  : '//localhost:8080')
  .concat(basePath);

const clientInfoPath = (data) => `${root}/clients/${data.componentClientId}/accounts/${data.accountId}/applications/${data.appId}`;
const projectInfo = ({ projectId }) => `projects/${projectId}`;
// TODO remove mockServer once APIs are implemented
// const runsInfo = ({ runId }) => `runs/${runId}/mockServer/`;
const runsInfo = ({ runId }) => `runs/${runId}`;
const clusterInfo = ({ clusterId }) => `clusters/${clusterId}`;
const transcriptInfo = ({ transcriptId }) => `conversations/${transcriptId}`;
const conversationTranscriptInfo = (transcriptId) => `conversations?tId=${transcriptId}`;

// TODO: Curation
// API: ....conversations?transcriptIds=['abc', 'def']&moveTo=<new cluster id>

export const apiUrls = {
  authentications: `${root}/authentications?self=true`,
  userConfigUrl: `${root}/user/config`,
  conversationsConfigUrl: `${root}/user/conversationsConfig`,
  answersConfigUrl: `${root}/user/answersConfig`,
  downloadTemplate: (data) => `${clientInfoPath(data)}/templates`,
  projects: (data) => `${clientInfoPath(data)}/projects`,
  oneProject: (data) => `${clientInfoPath(data)}/${projectInfo(data)}`,
  runs: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/runs`,
  oneRun: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}`,
  downloadRun: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/results?type=original`,
  clusters: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/clusters`,
  oneCluster: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}`,
  conversations: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/conversations`,
  transcript: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/${transcriptInfo(data)}`,
  intents: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/runs/${data.runId}/${clusterInfo(data)}/intents`,
  conversationTranscript: (data, transcriptId) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/${conversationTranscriptInfo(transcriptId)}`,
  interfaces: (data) => `${root}/clients/${data.componentClientId}/interfaces`,
  faqs: (data, interfaceId) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/interfaceId/${interfaceId}/addToAnswers`
};
