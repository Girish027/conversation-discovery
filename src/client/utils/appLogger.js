// log to console if method level meets log level
function appLogger(options) {
  // indexes of method names match values of levels properties
  const loggerMethods = ['trace', 'debug', 'info', 'warn', 'error'];
  const levels = {
    TRACE: 0,
    DEBUG: 1,
    INFO: 2,
    WARN: 3,
    ERROR: 4,
    SILENT: 5
  };
  const envLevels = {
    development: levels.TRACE,
    production: levels.SILENT,
    test: levels.ERROR
  };

  // environment variables can override default log level
  let logLevel = levels.INFO;
  if (typeof process.env.LOG_LEVEL !== 'undefined') {
    logLevel = levels[process.env.LOG_LEVEL];
  } else if (typeof process.env.NODE_ENV !== 'undefined') {
    logLevel = envLevels[process.env.NODE_ENV];
  }

  const logger = {};
  loggerMethods.forEach((methodName, methodLevel) => {
    logger[methodName] = (...args) => {
      // only print out if method level is not less than log level
      if (methodLevel >= logLevel) {
        let msgs = [];
        if (options && options.tag) {
          msgs.push(options.tag);
        }
        msgs = msgs.concat(args);
        /* eslint-disable no-alert, no-console */
        console.log(...msgs);
      }
    };
  });
  return logger;
}

export default appLogger;
