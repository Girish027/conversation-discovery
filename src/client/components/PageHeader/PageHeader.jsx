import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { fromJS } from 'immutable';
import { PageHeader, ToastNotification, StatusNotification } from '@tfs/ui-components';
import { getPageDetails } from 'utils/PageHeaderHelper';
import Constants from '../../constants';

class PageHeaderWrapper extends PureComponent {
  constructor(props) {
    super(props);

    this.getPageHeaderProps = this.getPageHeaderProps.bind(this);
  }

  getPageHeaderProps() {
    const {
      dispatch, caa, itsBaseUrl, routeName, activeProject, selectedRun,
    } = this.props;
    const { projectName, projectId, projectType } = activeProject.toJS();
    const { runName, runId } = selectedRun.toJS();

    const data = {
      dispatch,
      ...caa,
      itsBaseUrl,
      routeName,
      projectName,
      projectId,
      projectType,
      runName,
      runId
    };

    return getPageDetails(data);
  }

  render() {
    const { title = Constants.pageTitle, breadcrumb = [] } = this.getPageHeaderProps();
    const { header } = this.props;
    const { message, notificationType } = header;

    return (
      <PageHeader
        title={title}
        breadcrumb={breadcrumb}
        styleOverride={
          {
            secondaryNavBar: {
              paddingLeft: '30px'
            }
          }
        }
      >
        {(message && notificationType)
          && (
            <StatusNotification>
              <ToastNotification type={notificationType} styleOverride={{ top: '48px' }}>
                {message}
              </ToastNotification>
            </StatusNotification>
          )
        }
      </PageHeader>
    );
  }
}

PageHeaderWrapper.defaultProps = {
  activeProject: fromJS({}),
  selectedRun: fromJS({}),
  dispatch: () => {},
  itsBaseUrl: '',
  routeName: '',
  caa: {},
  header: {},
};

PageHeaderWrapper.propTypes = {
  caa: PropTypes.object,
  dispatch: PropTypes.func,
  routeName: PropTypes.string,
  itsBaseUrl: PropTypes.string,
  activeProject: PropTypes.object,
  selectedRun: PropTypes.object,
  header: PropTypes.object,
};

export default PageHeaderWrapper;
