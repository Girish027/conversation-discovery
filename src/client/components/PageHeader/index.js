import { connect } from 'react-redux';
import PageHeader from './PageHeader';
import { getPathnameSelector } from 'state/routerState';

import { getCAASelector, getITSBaseUrlSelector } from 'state/appState';
import { getActiveProject } from 'state/projectsState';
import { getSelectedRun } from 'state/runsState';
import { headerSelectors } from 'state/headerState';

const mapStateToProps = (state) => ({
  routeName: getPathnameSelector(state),
  caa: getCAASelector(state),
  itsBaseUrl: getITSBaseUrlSelector(state),
  activeProject: getActiveProject(state),
  selectedRun: getSelectedRun(state),
  header: headerSelectors.getNotifications(state),
});

const mapDispatchToProps = (dispatch) => ({
  dispatch
});

export default connect(mapStateToProps, mapDispatchToProps)(PageHeader);
