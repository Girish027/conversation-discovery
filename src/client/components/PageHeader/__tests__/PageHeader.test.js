import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ConnectedPageHeader from 'components/PageHeader';
import PageHeader from 'components/PageHeader/PageHeader';
import { fromJS } from 'immutable';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { routeNames } from 'utils/RouteHelper';

describe('<PageHeader />', function () {
  let wrapper;
  let store;

  const caa = {
    clientId: '247inc',
    appId: 'referencebot',
    accountId: 'referencebot',
  };

  const props = {
    caa,
    dispatch: () => {},
    routeName: routeNames.DISCOVER_INTENTS,
    itsBaseUrl: 'http://its.com',
    activeProject: fromJS({
      projectId: 'pro-abc',
      projectName: 'Project ABC'
    }),
    selectedRun: fromJS({
      runId: 'run-abc',
      runName: 'run ABC'
    }),
  };

  beforeAll(() => {
    // Mock store
    const mockStore = configureStore();
    store = mockStore(fromJS({
      app: {
        ...caa,
        itsBaseUrl: 'http://its.com',
      },
      projects: {
        projects: fromJS([
          {
            projectId: 'pro-123',
            projectName: 'Project 123'
          },
          {
            projectId: 'pro-abc',
            projectName: 'Project ABC'
          },
        ]),
        activeProjectId: 'pro-abc',
      },
      runs: {
        runs: fromJS([
          {
            runId: 'run-123',
            runName: 'run 123'
          },
          {
            runId: 'run-abc',
            runName: 'run ABC'
          },
        ]),
        selectedRunId: 'run-abc',
      },
      router: {
        action: 'PUSH',
        location: {
          pathname: routeNames.DISCOVER_INTENTS
        }
      }
    }));
  });


  afterAll(() => {
    jest.clearAllMocks();
  });

  const getConnectedWrapper = () => mount(
    <Provider store={store}>
      <ConnectedPageHeader />
    </Provider>);

  const getShallowWrapper = (propsObj) => shallow(<PageHeader
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist - connected component PageHeader', () => {
      wrapper = getConnectedWrapper();
      expect(wrapper.exists()).toBe(true);
    });

    test('should exist - basic component PageHeader', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly with correct title and breadcrumbs - dicoverIntent', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly with correct title and breadcrumbs - run overview', () => {
      wrapper = getShallowWrapper({ ...props, routeName: routeNames.DISTRIBUTION_GRAPH });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
