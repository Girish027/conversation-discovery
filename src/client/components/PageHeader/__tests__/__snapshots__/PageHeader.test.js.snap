// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> Snapshots renders correctly with correct title and breadcrumbs - dicoverIntent 1`] = `
<withTheme(PageHeader)
  breadcrumb={Array []}
  styleOverride={
    Object {
      "secondaryNavBar": Object {
        "paddingLeft": "30px",
      },
    }
  }
  title="Discover Intents"
/>
`;

exports[`<PageHeader /> Snapshots renders correctly with correct title and breadcrumbs - run overview 1`] = `
<withTheme(PageHeader)
  breadcrumb={
    Array [
      Object {
        "label": "Discover Intents",
        "onClick": undefined,
        "readOnly": true,
        "value": "Discover Intents",
      },
      Object {
        "accountId": "referencebot",
        "appId": "referencebot",
        "clientId": "247inc",
        "itsBaseUrl": "http://its.com",
        "label": "Project ABC",
        "onClick": [Function],
        "projectId": "pro-abc",
        "projectName": "Project ABC",
        "projectType": undefined,
        "readOnly": false,
        "routeName": "/conversationdiscovery/ui/run-overview",
        "runId": "run-abc",
        "runName": "run ABC",
        "value": "projectProject ABC",
      },
    ]
  }
  styleOverride={
    Object {
      "secondaryNavBar": Object {
        "paddingLeft": "30px",
      },
    }
  }
  title="run ABC Overview"
/>
`;
